src/ydata_synthetic/synthesizers/regular/cgan/model.py

```diff
@@ -136,9 +171,9 @@ def __init__(self, batch_size, num_classes):
 
     def build_model(self, input_shape, dim, data_dim):
         noise = Input(shape=input_shape, batch_size=self.batch_size)
-        label = Input(shape=(1,), batch_size=self.batch_size, dtype='int32')
-        label_embedding = Flatten()(Embedding(self.num_classes, 1)(label))
-        input = multiply([noise, label_embedding])
+        label = Input(shape=(self.num_classes,), batch_size=self.batch_size, dtype='int32')
+        label_embedding = Embedding(2, 1)(label)
```

Comment:
Not understanding the need to remove the flatten here. 

No changes should be done at the level of the generator. The processing was already done before, so nothing should change with the new processor. 

Please rever the Generator changes.